### ISSUE TITLE
Accessibility: Table's sortable headers can't be triggered with the keyboard

### ISSUE THREAD

**Issue by rezinak** [highcharts/highcharts#18519]
#### Expected behaviour
Sortable headers can be triggered from the keyboard

Ensure the column headers in the table are keyboard operable. A keyboard user must have the option to access the headers using the TAB key and sort the columns in the table.

Ensure all interactive functionality is operable with the keyboard. The best way to accomplish this is by using appropriate native controls, which come with keyboard functionality built in.

If using a native control is not possible, the control must have tabindex="0" and have JavaScript event listeners that allow the control to be activated when Enter is pressed.

#### Actual behaviour
The table headers in the table are not keyboard operable. The keyboard user cannot use the TAB key to navigate to the table headers

Live demo with steps to reproduce
https://www.idc.com/QA_test_section/accessibility-remediation/highcharts-demo1, view data table

Product version
Highcharts JS v10.3.3

Affected browser(s)
Chrome and Mozilla, haven't looked into the others

**Comment by oysteinmoseng** [comment]
Internal note: Should be fixed along with #18517

**Comment by jacob1474** [comment]
### Live Demo
[Link to live demo](https://jsfiddle.net/3z6e7L5u/)

### Steps to reproduce
Navigate to the chart using the keyboard (e.g., with the Tab key).

1. Press TAB to select series
2. Press TAB again to select legends
3. Press TAB again to select context menu
4. With Enter open context menu.
5. From the menu, select the option to view the Data Table with arrow keys.
6. Attempt to navigate to the clickable column headers within the data table using the keyboard.

### Expected Result
The user should be able to focus on each column header and activate its sorting function using the keyboard (e.g., by pressing Enter or Space).

### Actual Result
Keyboard focus skips the whole data table, making it impossible to access table headers to sort the data table without a mouse. These controls are only accessible via mouse click.

This behavior violates WCAG 2.1.1: Keyboard, as all interactive functionality must be operable through a keyboard.

### Affected versions
Highcharts v12.4

### Affected browsers
All tested browsers (e.g., Chrome, Firefox, Edge, Brave)
